### PR TITLE
Add Sciter 4.x port

### PR DESCRIPTION
**Pull request by pravic**
Based on https://github.com/c-smile/sciter-sdk, but Windows only.
Current version: 4.0.0.9.
Can be installed either as dynamic or static package (but static one requires the DLL presence still).

### CODE REVIEW

File: ports/sciter/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+set(SCITER_SRC ${CURRENT_BUILDTREES_DIR}/src/sciter-sdk-${SCITER_REVISION})
+
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL x64)
+  set(SCITER_ARCH 64)
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL x86)
+  set(SCITER_ARCH 32)
+endif()
+
+# unpack
+vcpkg_download_distfile(ARCHIVE
```
Comment (KindDragon) [inline ports/sciter/portfile.cmake]: Use `vcpkg_from_github` instead https://github.com/Microsoft/vcpkg/blob/05e682a19d40036c9a34c2c7a94e6325fb7e9f1c/ports/cpprestsdk/portfile.cmake#L3-L9

File: ports/sciter/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+endif()
+
+# bin
+if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+	file(INSTALL ${SCITER_SRC}/bin/${SCITER_ARCH}/sciter.dll DESTINATION ${CURRENT_PACKAGES_DIR}/bin)
+	file(INSTALL ${SCITER_SRC}/bin/${SCITER_ARCH}/sciter.dll DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
+	file(INSTALL ${SCITER_SRC}/bin/${SCITER_ARCH}/tiscript-sqlite.dll DESTINATION ${CURRENT_PACKAGES_DIR}/bin)
+	file(INSTALL ${SCITER_SRC}/bin/${SCITER_ARCH}/tiscript-sqlite.dll DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
+else()
+	message(WARNING "Sciter requires sciter.dll to run. Download it manually or install dynamic package.")
```
Comment (KindDragon) [inline ports/sciter/portfile.cmake]: Better use this condition on top of the file https://github.com/Microsoft/vcpkg/blob/05e682a19d40036c9a34c2c7a94e6325fb7e9f1c/ports/ace/portfile.cmake#L1-L4

Comment (KindDragon) [inline ports/sciter/portfile.cmake]: Also if library does not support static CRT linkage add this https://github.com/Microsoft/vcpkg/blob/05e682a19d40036c9a34c2c7a94e6325fb7e9f1c/ports/benchmark/portfile.cmake#L1-L3

**Comment by pravic** [comment]
Thanks for the review, I'll fix it.

File: ports/sciter/portfile.cmake (hunk 2 above)
Comment (KindDragon) [inline ports/sciter/portfile.cmake]: And please use 4 space indentation

File: ports/sciter/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,75 @@
+if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    message(WARNING "Warning: Sciter requires sciter.dll to run. Download it manually or install dynamic package.")
```
Comment (KindDragon) [inline ports/sciter/portfile.cmake]: Why not just exit with FATAL_ERROR?

Comment (pravic) [inline ports/sciter/portfile.cmake]: Because it is not a fatal error. You can build apps, you can run apps (if dll is already installed).

**Comment by ras0219-msft** [comment]
Hi @pravic, thanks for the PR, it looks great!

It's unfortunate that sciter publishes header files called "value.h" and "value.hpp" and also prefers being included without any subdirectories.

Despite that, I concur with your decision here to put the includes into a subdirectory 👍, it seems the best option.

**Comment by pravic** [comment]
Great, thanks!

**Pull request merged**